src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs

```diff
@@ -358,84 +340,76 @@ protected void UpdateActiveSourceRepository(string source, bool validateSource)
+            var resource = repository.GetResource<PackageSearchResource>();
+
+            return repository;
+        }
+
+        /// <summary>
+        /// Looks through all available sources (including those disabled) by matching source name and url to get matching sources.
+        /// </summary>
+        /// <param name="source">The source string specified by -Source switch.</param>
+        /// <returns>Returns an object of PackageSource if the specified source string is a known source. Else returns a null.</returns>
+        private PackageSource GetMatchingSource(string source)
+        {
+            var packageSources = _sourceRepositoryProvider.PackageSourceProvider?.LoadPackageSources();
+
+            var matchingSource = packageSources?.FirstOrDefault(
+                p => StringComparer.OrdinalIgnoreCase.Equals(p.Name, source) ||
+                     StringComparer.OrdinalIgnoreCase.Equals(p.Source, source));
+
+            return matchingSource;
+        }
+
+        /// <summary>
+        /// If a relative local URI is passed, it converts it into an absolute URI.
+        /// If the local URI does not exist or it is neither http nor local type, then the source is rejected.
         /// If the URI is not relative then no action is taken.
         /// </summary>
         /// <param name="source">The source string specified by -Source switch.</param>
-        /// <param name="packageId">The package ID string specified.</param>
-        /// <returns>Returns an absolute version of the source string if it is a relative local source. Else returns it unchanged.</returns>
-        /// <exception cref="PackageSourceException">Throws an PackageSourceException if the specified source is a non-existent local source or if the source is not http or local source.</exception>
-        protected string ConvertRelativeUriToAbsolute(string source)
+        /// <returns>The source validation result.</returns>
+        private SourceValidationResult CheckSourceValidity(string source, bool validateSource)
         {
+            // Convert a relative local URI into an absolute URI
             var packageSource = new PackageSource(source);
             Uri sourceUri;
             if (Uri.TryCreate(source, UriKind.Relative, out sourceUri))
             {
                 string outputPath;
                 bool? exists;
                 string errorMessage;
-                if (TryTranslatePSPath(source, out outputPath, out exists, out errorMessage)
-                    && exists == true)
+                if (PSPathUtility.TryTranslatePSPath(SessionState, source, out outputPath, out exists, out errorMessage) &&
+                    exists == true)
                 {
                     source = outputPath;
+                    packageSource = new PackageSource(source);
                 }
-                else if (exists.HasValue && !exists.Value)
+                else if (exists == false)
```

Comment:
Correct, which would fail this check. `exists.HasValue && !exists.Value` is equivalent to `exists == false`.